### PR TITLE
DCOS-20689: introduce Sentry bug tracking SaaS

### PR DESCRIPTION
**Pull request by nLight**
Track errors with [Sentry](https://sentry.io)

Closes DCOS-20689

## Testing

- Enable `tracking` plugin in your Config.dev.ts
- Add an exception to any place in code
- Reach this place on the UI
- Verify that the error is reflected and all tags (environment, variant, version) are set.

## Trade-offs

We track two versions: DCOS and DCOS-UI. DCOS-UI version will be set as a `release` in Sentry where dcos version as a tag. I did this because we are going to be shipping UI as a package thus having the same UI release running on different DCOS versions. Plus I'm going to connect Sentry and Github to provide better dev experience.

### CODE REVIEW

File: plugins/tracking/hooks.js [hunk 1]
```diff
@@ -59,6 +67,37 @@ module.exports = {
     return promiseArray;
   },
 
+  pluginsConfigured() {
+    // Ensure analytics is actually ready, because in #pluginsLoadedCheck we
+    // may skip the check so that we don't completely block the applicaiton
+    const metadata =
+      Util.findNestedPropertyInObject(this, "configuration.metadata") || {};
+
+    global.analytics.ready(() => {
```
Comment (jaxesn) [inline plugins/tracking/hooks.js]: I think i've asked this before, but where does `global.analytics` come from?

Comment (nLight) [reply plugins/tracking/hooks.js]: It comes from segment.io that we're using to inject all other services.

**Comment by nLight** [comment]
WIP: turned out because of the current `semantic-release` setup we don't have release version in `package.json` our build/release process has to be adjusted.

**Comment by juliangieseke** [comment]
I did a local test which was able to replace the placeholder and echo the upload command.

---

updated `dist/index.html`:
```
<script type="text/javascript"> window.DCOS_UI_VERSION = "master+v2.14.0"; </script>
```
**=> is the release name correct? yes / no**

---

`aws s3 cp` upload cmd:
```
aws s3 cp …/release.tar.gz s3://…/master+dcos-ui-v2.14.0.tar.gz
```
**=> is the filename correct? yes / no**

---

Once CI is green we should be able to merge - after we merged, we should keep an eye on the build to verify that everything actually works as expected 🤷‍♂️  (if not, its not _that_ bad - worst case is that our daily FE cluster is broken - we cant break anything _really_ important)

File: scripts/ci/upload-release [hunk 2]
```diff
@@ -32,8 +32,8 @@ PKG_TARGET=$(python -c "import sys, json; \
   dcosrc = json.load(open('$PROJECT_ROOT/.dcosrc')); \
   sys.stdout.write(str(dcosrc['base']));")
 
-# release version (after switching to semantic-release, we cant use package.json anymore)
-PKG_VERSION="$(git tag -l --sort=-version:refname | grep ${PKG_TARGET} | head -1 | cut -d'+' -f2)"
+# next release version (passed in by semantic-release)
+PKG_VERSION="${1}"
```
Comment (juliangieseke) [inline scripts/ci/upload-release]: version is now injected by semanric-release ->  https://github.com/dcos/dcos-ui/pull/3192/files#diff-b78e9a225940bc2471e85d5d79220450R9

File: scripts/ci/upload-release [hunk 3]
```diff
@@ -47,6 +47,14 @@ DCOS_BRANCH="dcos-ui/${PKG_TARGET}+dcos-ui-latest"
 ## Create Assets & Generate Upload Data
 #####################################################################
 
+
+# first, replace placeholder in dist bundle
+sed -i "" "s/0.0.0-dev+semantic-release/${PKG_TARGET}+${PKG_VERSION}/" ${PROJECT_ROOT}/dist/index.html
```
Comment (juliangieseke) [inline scripts/ci/upload-release]: here we inject the new version into `dist/index.html`

File: scripts/ci/upload-release [hunk 4]
```diff
@@ -47,6 +47,19 @@ DCOS_BRANCH="dcos-ui/${PKG_TARGET}+dcos-ui-latest"
 ## Create Assets & Generate Upload Data
 #####################################################################
 
+VERSION_PLACEHOLDER="0.0.0-dev+semantic-release"
```
Comment (nLight) [inline scripts/ci/upload-release]: @DanielMSchmidt added

Comment (juliangieseke) [reply scripts/ci/upload-release]: we could read it from `package.json` as we do with `PKG_TARGET` and `.dcosrc` above :trollface:

File: plugins/tracking/hooks.js [hunk 5]
```diff
@@ -59,6 +67,37 @@ module.exports = {
...
+      const setContext = () => {
+        if (!global.Raven) {
+          return;
+        }
+
+        global.Raven.setTagsContext({
+          ...metadata,
+          environment: process.env.NODE_ENV,
+          dcosVersion: MetadataStore.version,
```
Comment (weblancaster) [inline plugins/tracking/hooks.js]: @nLight on line `90` you are checking for `!MetadataStore.version` but not here, so I wonder if you are ok in having a falsey `dcosVersion`?

**Pull request merged**